### PR TITLE
Upgrade ucx from 1.11.0 to 1.15.0

### PR DESCRIPTION
**Pull request by jcamposeco**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Update ucx packate to v1.15.0

###### Change Log  

- ucx package upgraded

###### Does this affect the toolchain?  

NO

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=491572&view=results

### CODE REVIEW

File: SPECS/ucx/ucx.signatures.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
  "Signatures": {
-  "ucx-1.11.0.tar.gz": "b7189b69fe0e16e3c03784ef674e45687a9c520750bd74a45125c460ede37647"
+  "ucx-1.11.0.tar.gz": "4b202087076bc1c98f9249144f0c277a8ea88ad4ca6f404f94baa9cb3aebda6d"
```
Comment (anphel31) [inline SPECS/ucx/ucx.signatures.json:3]: this filename should change

File: SPECS/ucx/ucx.spec [hunk 2]
```diff
@@ -8,23 +8,28 @@
...
 %bcond_with    ugni
 %bcond_with    xpmem
+%bcond_with    vfs
 
 Summary:        UCX is a communication library implementing high-performance messaging
 Name:           ucx
-Version:        1.11.0
-Release:        4%{?dist}
+Version:        1.15.0
+Release:        5%{?dist}
```
Comment (anphel31) [inline SPECS/ucx/ucx.spec:16]: you can set the release to `1`, since the version is being incremented

**Review by anphel31 — COMMENTED** [review]
Can you please update cgmanifest.json as well with the updated URL

File: SPECS/ucx/ucx.signatures.json (hunk 1 above)
Comment (jcamposeco) [reply SPECS/ucx/ucx.signatures.json:3]: right, changed now

File: SPECS/ucx/ucx.spec [hunk 3]
```diff
@@ -8,23 +8,28 @@
...
 
 # UCX currently supports only the following architectures
 ExclusiveArch: aarch64 ppc64le x86_64
 
-BuildRequires: automake autoconf libtool gcc-c++ numactl-devel
+%if %{defined extra_deps}
+Requires: %{?extra_deps}
+%endif
+
+BuildRequires: automake autoconf libtool gcc-c++
```
Comment (anphel31) [inline SPECS/ucx/ucx.spec:32]: can you put these on separate lines to make them easier to read?

File: SPECS/ucx/ucx.signatures.json (hunk 1 above)
Comment (mfrw) [reply SPECS/ucx/ucx.signatures.json]: Giving our make command line an option of `SRPM_FILE_SIGNATURE_HANDLING=update` updates the `*.signatures.json` file automatically .. 
FYI .. Just incase you might not be aware :)

File: SPECS/ucx/ucx.spec [hunk 4]
```diff
@@ -8,23 +8,28 @@
...
 %if %{with cma}
 BuildRequires: glibc-devel >= 2.15
 %endif
```
Comment (anphel31) [inline SPECS/ucx/ucx.spec:35]: Can we also get rid of the dependency on `knem` (which is currently not buildable on 3.0-dev?

File: SPECS/ucx/ucx.spec (hunk 3 above)
Comment (jcamposeco) [reply SPECS/ucx/ucx.spec:32]: done

**Review by anphel31 — APPROVED** [review]
can you please update the description with a link to the buddy build

**Pull request merged**